### PR TITLE
Adds basic dependency injection support to the project.

### PR DESCRIPTION
**Pull request by tacosontitan**

### CODE REVIEW

**Comment by tacosontitan** [comment]
I apologize, I didn't realize this PR was still open. I'll create a branch for future changes until you get a chance to review and complete this one.

This PR effectively adds support for dependency injection, defines a game service, and adds support for VS code debugging. Please let me know if you have any questions, I'll gladly answer them in detail!

**Comment by lukkasz323** [comment]
@tacosontitan It'll take make some time as I haven't worked with services before. What is its purpose in this case? Thanks.

**Comment by tacosontitan** [comment]
## 💁‍♀️ Why dependency injection?
The biggest advantage dependency injection provides is decoupling the usage of an object from its creation which promotes single responsibility among other SOLID principles.

## 🎮 What is `GameService`?
This object is a simple implementation of `IHostedService` that focuses solely on managing your `Game` instance. You'll notice that it simply calls `Game.Run` and ensures that the service is properly stopped by handling `Game.Exiting`.

## ℹ️ Why `IHostedService`?
I created `GameService` after setting up the host so that handling the `Game` instance could happen in a dedicated object while allowing it to run independently of other potentially long-running services your project might need. In addition to this, the host knows when it's safe to start a hosted service and in fact, starts it for you when it is safe to do so; as a result, I decided not to try and fight between the `Game.Run` invocation and running the host, both of which I believe are blocking calls, if I remember correctly.

**Comment by tacosontitan** [comment]
TLDR; Setting up dependency injection provides a greater benefit to the rest of the project as a whole, so using a hosted service for managing the `Game` instance seemed to be the way to go for a proper setup.

**Comment by lukkasz323** [comment]
Seems great, I'll learn more about Logging / DependencyInjection MS extensions and I'll merge this if there won't be any problems. I've never merged PRs before, I'm a few commits ahead locally, so I believe I'll have to rebase this PR and resolve conflicts? Not sure yet.

**Comment by tacosontitan** [comment]
> I've never merged PRs before, I'm a few commits ahead locally, so I believe I'll have to rebase this PR and resolve conflicts? Not sure yet.

If you prefer, you can push your changes and I'll update my branch prior to your merge operation so that it's already up-to-date.

I have another set of changes ready to go related to the input systems.

**Comment by tacosontitan** [comment]
The merge process forced me to abandon these changes. However, I can create a new pull request shortly after resolving the conflicts on my side.
